Feature: Handle turn on/off all intent

Tony763 opened the pull request:
- Added handler for turning on/off all entities in switch ar light group
- Address PR #38
- Address Issue #20

## Description:


## Checklist:
  - [ ] Code is Python 3.x compatible
  - [ ] Travis is passing tests **Your PR cannot be merged unless tests pass**
  - [ ] No PEP Bot Errors or Issues

Tony763 committed:
Feature: Handle turn on/off all intent

- Added handler for turning on/off all entities in switch ar light group
- Address PR #38

stratus-ss commented on __init__.py:
```diff
@@ -199,6 +199,28 @@ def _handle_switch(self, message):
         LOGGER.debug("Entity: %s" % entity)
         LOGGER.debug("Action: %s" % action)
 
```
There was a separate logger PR but it appears this got missed

Tony763 replied:
Updated with upstream

Tony763 committed:
Merge remote-tracking branch 'upstream/20.08' into patch-5

fermulator commented on __init__.py:
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
         self.log.debug("Entity: %s" % entity)
         self.log.debug("Action: %s" % action)
 
+        # Handle turn on/off all intent
```
if this handles both switch and light domains, should this code really live in `_handle_switch`?

it feels like "something" isn't right, if we're setting domain away from `switch` in this method

Tony763 replied:
OK, renamed

fermulator commented on __init__.py:
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
+            else:
+                domain = None
+
+            if not domain is None:
```
`if domain is not None` is the preferred way to write this

```
>>> foo = None
>>> if foo is not None:
...   print('fi')
... 
>>> foo = 'hi'
>>> if foo is not None:
...   print('bye')
... 
bye
```

it is more readable

Tony763 replied:
OK

fermulator commented on __init__.py:
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
+            elif self.voc_match(entity,"all_switches"):
+                domain = "switch"
+            else:
+                domain = None
```
no handling/response currently for this case (silent failure)

fermulator replied:
should we log / say something, then return?

Tony763 replied:
There is nothing wrong, You decide if intent is for all lights or switches. If it not match, then intent is not for all lights/switches and You continue with single entity. (Starting on line 220)

fermulator commented on __init__.py:
```diff
@@ -197,6 +197,28 @@ def _handle_switch(self, message):
+                self.ha.execute_service(domain, "turn_%s" % action, ha_data)
+                self.speak_dialog('homeassistant.device.%s' % action, data=ha_entity)
+                return
+        # TODO: need to figure out, if this indeed throws a KeyError
```
if we're not sure, or it is too difficult to isolate, we could catch for generic `Exception`?

Tony763 replied:
OK

Commit:
Requested changes

Renamed '_handle_switch' to '_handle_turn_actions'
Corrected 'if domain is not None' format
Added generic exception catching for turn all intent handler

Tony763 commented:
@fermulator  Could You re-review?

stratus-ss reviewed (approved):
LGTM
